ray_on_golem/reputation/plugins/score.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from typing import TYPE_CHECKING, Dict, Final, Optional, Sequence, Tuple
+
+from golem.managers.base import ProposalScorer, ProposalScoringResult
+from golem.utils.typing import MaybeAwaitable
+from tortoise.exceptions import DoesNotExist
+
+from ray_on_golem.reputation import models as m
+
+if TYPE_CHECKING:
+    from golem.resources import ProposalData
+
+logger = logging.getLogger(__name__)
+
+PARAM_UPTIME = "uptime"
+PARAM_SUCCESS_RATE = "success_rate"
+DEFAULT_UPTIME_WEIGHT: Final[float] = 1.0
+DEFAULT_SUCCESS_RATE_WEIGHT: Final[float] = 0.5
+
+
+class ReputationScorer(ProposalScorer):
+    def __init__(
+        self,
+        payment_network: str,
+        weights: Optional[Dict[str, float]] = None,
+        score_range: Tuple[float, float] = (0.0, 1.0),
+    ):
+        self.payment_network = payment_network
+        self.weights = weights or self._default_weights()
+        self.total_weight = sum(self.weights.values())
+        self.score_range = score_range
+        logger.debug(
+            f"Initialized {self.__class__.__name__}: "
+            "payment_network=%s, weights=%s, score_range=%s",
+            self.payment_network,
+            self.weights,
+            self.score_range,
+        )
+
+    @staticmethod
+    def _default_weights() -> Dict[str, float]:
+        return {
+            PARAM_UPTIME: DEFAULT_UPTIME_WEIGHT,
+            PARAM_SUCCESS_RATE: DEFAULT_SUCCESS_RATE_WEIGHT,
+        }
+
+    async def calculate_score(
+        self,
+        proposal_data: "ProposalData",
+    ) -> float:
+        provider_id = proposal_data.issuer_id
+
+        try:
+            node_reputation = await m.NodeReputation.get_blacklisted(False).get(
+                network__network_name=self.payment_network, node__node_id=provider_id
+            )
+        except DoesNotExist:
+            node_reputation = m.NodeReputation()
+            logger.debug("Provider `%s` not found. Assuming zero scores.", provider_id)
```

Comment:
If scorers support `None` why not to use `None` instead of `0`?

Reply:
hmm:
1. what does a score of `None` mean?
2. I believe that having no reputation _should_ be prejudiced against, as providers _should_ be interested in having the best opinion and it doesn't sound fair to promote starting from scratch e.g. to fix your reputation score...

Reply:
1. By looking at the `ProposalScoringMixin` it looks like it is not supported anymore. @approxit do you recall what was meant to happen in that scenario?
2. Fair point